### PR TITLE
LocaleSetting: add search to dropdowns

### LINKED ISSUE #70: Formats dropdown is slow to navigate.
scrolling down or up the list isn't intuitive, and is slow because are no scrollbar or obvious search methods. You have to hover over the upper or lower section of the list for it to start moving. Scrolling with the scrollwheel moves the list by half an item per tick.

### PR DESCRIPTION
**Pull request by danirabbit**
Fixes #70 

And use expression instead of factory

### CODE REVIEW

File: src/Widgets/LocaleSetting.vala [hunk 1]
```diff
@@ -41,21 +41,26 @@ namespace SwitchboardPlugLocale.Widgets {
...
-            region_factory.setup.connect (region_setup_factory);
-            region_factory.bind.connect (region_bind_factory);
+            var expression = new Gtk.CClosureExpression (
+                typeof (string),
+                null,
+                {},
+                (Callback) get_locale_name,
+                null,
+                null
+            );
```
Comment (jeremypw) [inline src/Widgets/LocaleSetting.vala:51]: Simpler to use a Gtk.PropertyExpression:
```
            var expression = new Gtk.PropertyExpression (
                typeof (Locale),
                null,
                "name"
            );
```

**Review by jeremypw — COMMENTED** [review]
I suggest using a PropertyExpression - do not need a callback then.

File: src/Widgets/LocaleSetting.vala (hunk 1 above)
Comment (danirabbit) [reply src/Widgets/LocaleSetting.vala:51]: That's awesome thank you! I'm still learning how to use the Expressions stuff so that's great. Appreciate it!

Comment (jeremypw) [reply src/Widgets/LocaleSetting.vala:51]: Me too!

**Review by jeremypw — APPROVED** [review]
Works as expected, code good.

**Pull request merged**